src/vector.js

```diff
@@ -18,12 +18,10 @@ export default class Vector extends MathArray {
 
   lengthSquared() {
     let length = 0;
-    if (length !== 0) {
-      for (let i = 0; i < this.ELEMENTS; ++i) {
-        length += this[i] * this[i];
-      }
+    for (let i = 0; i < this.ELEMENTS; ++i) {
+      length += this[i] * this[i];
     }
-    return checkNumber(this);
+    return length;
```

Comment:
checkNumber(length)?

Reply:
The components of a vector is already checked. We only need to call `checkNumber` when there's a passed-in argument involved.